otodom/task_1/mk/otodom.py

```diff
@@ -0,0 +1,165 @@
+import json
+import re
+import sys
+
+import requests
+from bs4 import BeautifulSoup
+
+HEADERS = {
+    "Access-Control-Allow-Origin": "*",
+    "Content-Type": "text/html; charset=utf-8",
+    "Accept-Language": "pl-PL,pl;q=0.9,en-US;q=0.8,en;q=0.7",
+    "Sec-Ch-Ua": '"Chromium";v="118", "Google Chrome";v="118", "Not=A?Brand";v="99"',
+    "User-Agent": "Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/118.0.0.0 Safari/537.36",  # noqa: E501
+}
+
+
+class Crawler:
+    def getPage(self, url):
+        try:
+            req = requests.get(url, headers=HEADERS)
+        except requests.exceptions.RequestException:
+            print(f"Failed to retrieve the webpage. Status code: {req.status_code}")
+            return None
+        return BeautifulSoup(req.text, "html.parser")
+
+    def getLinks(self, url):
+        bs = self.getPage(url)
+        if bs is not None:
+            promoted_listings = bs.find(
+                "div", {"data-cy": "search.listing.promoted"}
+            ).find_all("a", {"class": "css-1hfdwlm e1dfeild2"})
+            promoted = []
+            for link in promoted_listings:
+                if "href" in link.attrs:
+                    promoted.append(link.attrs["href"])
+
+            organic_listings = bs.find(
+                "div", {"data-cy": "search.listing.organic"}
+            ).find_all("a", {"class": "css-1hfdwlm e1dfeild2"})
+            organic = []
+            for link in organic_listings:
+                if "href" in link.attrs:
+                    organic.append(link.attrs["href"])
+
+            return (promoted, organic)
+        return None
+
+    def getListing(self, pageUrl, promoted):
+        url = "http://www.otodom.pl{}".format(pageUrl)
+        bs = self.getPage(url)
+        if bs is not None:
+            listing = dict()
+            listing["url"] = url
+
+            id_match = re.search(
+                "[0-9]+$", bs.find("meta", {"name": "description"}).get("content", "")
+            )
+            listing["otodom_id"] = id_match.group() if id_match else ""
+
+            listing["title"] = bs.find("h1", {"class": "css-1wnihf5 efcnut38"}).text
+
+            localization = dict()
+            l10n = bs.find("a", {"class": "e1w8sadu0 css-1helwne exgq9l20"}).text.split(
+                ","
+            )
+
+            localization["province"] = l10n[-1] if len(l10n) >= 4 else ""
+            localization["city"] = l10n[-2] if len(l10n) >= 4 else l10n[-1]
+            localization["district"] = l10n[1] if len(l10n) >= 4 else ""
+            localization["street"] = l10n[0] if len(l10n) >= 4 else l10n[0]
```

Comment:
you can use `match` on the `len(l10n)` and dependignly on case do something. I think it will be a more readable